stablehlo/reference/Ops.cpp

```diff
@@ -314,6 +325,147 @@ Tensor evalRealOp(const Tensor &operand, TensorType resultType) {
   return result;
 }
 
+SmallVector<Tensor> evalReduceOp(ArrayRef<Tensor> inputs,
+                                 ArrayRef<Tensor> initValues,
+                                 const Axes &dimensions, Region &body,
+                                 Scope &scope,
+                                 ArrayRef<TensorType> resultTypes) {
+  SmallVector<Tensor> results;
+  for (auto [resultType, initValue] : llvm::zip(resultTypes, initValues)) {
+    Tensor result = Tensor(resultType);
+    for (auto resultIt = result.index_begin(); resultIt != result.index_end();
+         ++resultIt)
+      result.set(*resultIt, initValue.get({}));
+    results.push_back(result);
+  }
+  for (auto inputIt = inputs[0].index_begin(); inputIt != inputs[0].index_end();
+       ++inputIt) {
+    Index resultIndex = *inputIt;
+    for (int64_t dim : llvm::reverse(dimensions))
+      resultIndex.erase(resultIndex.begin() + dim);
+    SmallVector<Tensor> args;
+    for (auto [runtimeResult, initValue] : llvm::zip(results, initValues)) {
+      auto arg = Tensor(initValue.getType());
+      arg.set({}, runtimeResult.get(resultIndex));
+      args.push_back(arg);
+    }
+    for (auto [input, initValue] : llvm::zip(inputs, initValues)) {
+      auto arg = Tensor(initValue.getType());
+      arg.set({}, input.get(*inputIt));
+      args.push_back(arg);
+    }
+    SmallVector<Tensor> reducedValues = eval(body, args, &scope);
+    for (auto [result, value] : llvm::zip(results, reducedValues))
+      result.set(resultIndex, value.get({}));
+  }
+  return results;
+}
+
+SmallVector<Tensor> evalReduceWindowOp(
+    ArrayRef<Tensor> inputs, ArrayRef<Tensor> initValues,
+    const Axes &windowDimensions, const Axes &windowStrides,
+    const Axes &baseDilations, const Axes &windowDilations,
+    const SmallVector<SmallVector<int64_t>> &padding, Region &body,
+    Scope &scope, ArrayRef<TensorType> resultTypes) {
+  SmallVector<Tensor> runtimeResults;
+  Builder builder = mlir::Builder(resultTypes[0].getContext());
+  Type i64Type = builder.getI64Type();
+
+  for (auto [resultType, initValue] : llvm::zip(resultTypes, initValues)) {
```

Comment:
It seems that initializing a Tensor with some value is a frequent operation as well. How about we add a constructor:
```
explicit Tensor(TensorType type, const Element &element);
```